components/clp-package-utils/clp_package_utils/scripts/archive_manager.py

```
@@ -0,0 +1,237 @@
+        FIND_COMMAND == subcommand
+    ):
+        begin_timestamp = parsed_args.begin_ts
+        end_timestamp = parsed_args.end_ts
+
+        # Validate the input timestamp
+        assert begin_timestamp is not None, "begin_timestamp is None."
+        if not _validate_timestamps(begin_timestamp, end_timestamp):
+            return -1
+
+    container_name: str = generate_container_name("archive-manager")
+
+    container_clp_config, mounts = generate_container_config(clp_config, clp_home)
+    generated_config_path_on_container, generated_config_path_on_host = dump_container_config(
+        container_clp_config, clp_config, container_name
+    )
+
+    necessary_mounts: typing.List[CLPDockerMounts] = [
+        mounts.clp_home,
+        mounts.logs_dir,
+        mounts.archives_output_dir,
+    ]
+    container_start_cmd: typing.List[str] = generate_container_start_cmd(
+        container_name, necessary_mounts, clp_config.execution_container
+    )
+
+    # fmt: off
+    archive_manager_cmd: typing.List[str] = [
+        "python3",
+        "-m", "clp_package_utils.scripts.native.archive_manager",
+        "--config", str(generated_config_path_on_container),
+        str(subcommand),
+    ]
+    # fmt : on
+
+    # Add subcommand-specific arguments
+    if DEL_COMMAND == subcommand:
+        if parsed_args.dry_run:
+            archive_manager_cmd.append(DRY_RUN_ARG)
+        if DEL_BY_IDS_SUBCOMMAND == parsed_args.del_subcommand:
+            archive_manager_cmd.append(DEL_BY_IDS_SUBCOMMAND)
+            archive_manager_cmd.extend(parsed_args.ids)
+        elif DEL_BY_FILTER_SUBCOMMAND == parsed_args.del_subcommand:
+            archive_manager_cmd.extend([
+                DEL_BY_FILTER_SUBCOMMAND,
+                str(begin_timestamp),
+                str(end_timestamp)
+            ])
+        else:
+            logger.error(f"Unsupported subcommand: `{parsed_args.del_subcommand}`.")
```

If we have unsupported subcommands, shall we return an error immediately?

---

Right, we should

---

the argparse should reject invalid commands/arguments, I think this is just in case there's something added later on but not done properly.